plugins/module_utils/common/argspec_validate.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""Use AnsibleModule's argspec validation
+
+def _check_argspec(self):
+    aav = AnsibleArgSpecValidator(
+        data=self._task.args,
+        schema=DOCUMENTATION,
+        schema_format="doc",
+        schema_conditionals={},
+        name=self._task.action,
+    )
+    valid, errors = aav.validate()
+    if not valid:
+        raise AnsibleActionFail(errors)
+
+"""
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    dict_merge,
+)
+from ansible.module_utils.six import iteritems, string_types
+from ansible.module_utils._text import to_bytes
+
+try:
+    from packaging.specifiers import SpecifierSet
```

Comment:
Why is this import required?

Reply:
It's not, left over.  I was originally using the ansible version to determine is the future ansible-base functionality should be used once it's out of AnsibleModule.  In the end I went with a try/except instead.  good catch